src/utils/react-fast-compare.ts

```diff
@@ -1,75 +1,69 @@
     if (Array.isArray(a)) {
-      length = a.length;
-      if (length != b.length) return false;
-      for (i = length; i-- !== 0;)
-        if (!equal(a[i], b[i])) return false;
-      return true;
+      length = a.length
+      if (length !== b.length) return false
+      for (i = length; i-- !== 0; ) if (!equal(a[i], b[i])) return false
+      return true
     }
 
-    var it;
+    var it
     if (typeof Map === 'function' && a instanceof Map && b instanceof Map) {
-      if (a.size !== b.size) return false;
-      it = a.entries();
-      while (!(i = it.next()).done)
-        if (!b.has(i.value[0])) return false;
-      it = a.entries();
-      while (!(i = it.next()).done)
-        if (!equal(i.value[1], b.get(i.value[0]))) return false;
-      return true;
+      if (a.size !== b.size) return false
+      it = a.entries()
+      while (!(i = it.next()).done) if (!b.has(i.value[0])) return false
+      it = a.entries()
+      while (!(i = it.next()).done) if (!equal(i.value[1], b.get(i.value[0]))) return false
+      return true
     }
 
     if (typeof Set === 'function' && a instanceof Set && b instanceof Set) {
-      if (a.size !== b.size) return false;
-      it = a.entries();
-      while (!(i = it.next()).done)
-        if (!b.has(i.value[0])) return false;
-      return true;
+      if (a.size !== b.size) return false
+      it = a.entries()
+      while (!(i = it.next()).done) if (!b.has(i.value[0])) return false
+      return true
     }
 
-    if (a.constructor === RegExp) return a.source === b.source && a.flags === b.flags;
-    if (a.valueOf !== Object.prototype.valueOf) return a.valueOf() === b.valueOf();
-    if (a.toString !== Object.prototype.toString) return a.toString() === b.toString();
+    if (a.constructor === RegExp) return a.source === b.source && a.flags === b.flags
+    if (a.valueOf !== Object.prototype.valueOf) return a.valueOf() === b.valueOf()
+    if (a.toString !== Object.prototype.toString) return a.toString() === b.toString()
 
-    keys = Object.keys(a);
-    length = keys.length;
-    if (length !== Object.keys(b).length) return false;
+    keys = Object.keys(a)
+    length = keys.length
+    if (length !== Object.keys(b).length) return false
 
-    for (i = length; i-- !== 0;)
-      if (!Object.prototype.hasOwnProperty.call(b, keys[i])) return false;
+    for (i = length; i-- !== 0; ) if (!Object.prototype.hasOwnProperty.call(b, keys[i])) return false
 
-    if (typeof Element !== 'undefined' && a instanceof Element) return false;
+    if (typeof Element !== 'undefined' && a instanceof Element) return false
 
-    for (i = length; i-- !== 0;) {
-      if (keys[i] === '_owner' && a.$$typeof) continue;
-      if (!equal(a[keys[i]], b[keys[i]])) return false;
+    for (i = length; i-- !== 0; ) {
+      if (keys[i] === '_owner' && a.$$typeof) continue
+      if (!equal(a[keys[i]], b[keys[i]])) return false
     }
-    return true;
+    return true
   }
 
-  return a !== a && b !== b;
```

Comment:
That's incorrect. It handles NaN case, NaN !== NaN is true, so it says that both a and b should be NaN.